core/src/test/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbeTest.java

```diff
@@ -148,6 +153,60 @@ void shouldRecognizeInvalidGitHubUrl() throws Exception {
         final ProbeResult r1 = probe.apply(p1, ctx);
 
         assertThat(r1.status()).isEqualTo(ResultStatus.FAILURE);
-        assertThat(r1.message()).isEqualTo("The plugin SCM link is invalid");
+        assertThat(r1.message()).isEqualTo("The plugin SCM link is invalid.");
+    }
+
+    @Test
+    void shouldReturnCorrectScmFolderPath() throws IOException {
+        final Plugin plugin = mock(Plugin.class);
+        final GitHub github = mock(GitHub.class);
+        final String repositoryName = "jenkinsci/test-repo";
+        ProbeContext ctx = new ProbeContext(plugin.getName(), new UpdateCenter(Map.of(), Map.of(), List.of()));
```

Comment:
Why do you need this one?

Reply:
Do you mean this line?

>         ProbeContext ctx = new ProbeContext(plugin.getName(), new UpdateCenter(Map.of(), Map.of(), List.of()));

I initliaized this to use `spy`.

Do you suggest I do it the following way:

`ProbeContext contextSpy = spy(new ProbeContext(plugin.getName(), new UpdateCenter(Map.of(), Map.of(), List.of())));`?